packages/permissionless/accounts/light/signerToLightSmartAccount.ts

```diff
@@ -0,0 +1,439 @@
+            )
+        },
+        signTransaction: (_, __) => {
+            throw new SignTransactionNotSupportedBySmartAccount()
+        },
+        async signTypedData<
+            const TTypedData extends TypedData | Record<string, unknown>,
+            TPrimaryType extends
+                | keyof TTypedData
+                | "EIP712Domain" = keyof TTypedData
+        >(typedData: TypedDataDefinition<TTypedData, TPrimaryType>) {
+            return signWith1271WrapperV1<TSource, TAddress>(
+                signer,
+                chainId,
+                accountAddress,
+                hashTypedData(typedData)
+            )
+        },
+        client: client,
+        publicKey: accountAddress,
+        entryPoint: entryPointAddress,
+        source: "LightSmartAccount",
+        async getNonce() {
+            return getAccountNonce(client, {
+                sender: accountAddress,
+                entryPoint: entryPointAddress
+            })
+        },
+        async signUserOperation(userOperation) {
+            return signMessage(client, {
+                account: viemSigner,
+                message: {
+                    raw: getUserOperationHash({
+                        userOperation,
+                        entryPoint: entryPointAddress,
+                        chainId: chainId
+                    })
+                }
+            })
+        },
+        async getInitCode() {
+            if (smartAccountDeployed) return "0x"
+
+            smartAccountDeployed = await isSmartAccountDeployed(
+                client,
+                accountAddress
+            )
+
+            if (smartAccountDeployed) return "0x"
+
+            return concatHex([
+                factoryAddress,
+                await getAccountInitCode(viemSigner.address, index)
+            ])
+        },
+        async getFactory() {
+            if (smartAccountDeployed) return undefined
+            smartAccountDeployed = await isSmartAccountDeployed(
+                client,
+                accountAddress
+            )
+            if (smartAccountDeployed) return undefined
+            return factoryAddress
+        },
+        async getFactoryData() {
+            if (smartAccountDeployed) return undefined
+            smartAccountDeployed = await isSmartAccountDeployed(
+                client,
+                accountAddress
+            )
+            if (smartAccountDeployed) return undefined
+            return getAccountInitCode(viemSigner.address, index)
+        },
+        async encodeDeployCallData(_) {
+            throw new Error("Simple account doesn't support account deployment")
+        },
+        async encodeCallData(args) {
+            if (Array.isArray(args)) {
+                const argsArray = args as {
+                    to: Address
+                    value: bigint
+                    data: Hex
+                }[]
+
+                if (getEntryPointVersion(entryPointAddress) === "v0.6") {
```

Comment:
why is this dependent ion entryPoint version? both entryPoint version can call any of the wallet's execute implementation.